pandas/core/internals/managers.py

```diff
@@ -688,8 +689,9 @@ def get_bool_data(self, copy: bool = False) -> "BlockManager":
         copy : bool, default False
             Whether to copy the blocks
         """
-        self._consolidate_inplace()
```

Comment:
I think consolidating here might be the better option, as it at least ensures consistent behaviour when you have multiple columns independent of consolidation status. The inconsistency between a single column and multiple column is still present of course, but IMO there is nothing to do about this giving our consolidated blocks (well, we could deprecate object dtype being regarded as bool ..)

Reply:
> well, we could deprecate object dtype being regarded as bool 

This PR does that (well changes outright, not deprecates).  AFAICT thats the only way to make the behavior independent of whether the presence of another object-dtype-but-not-bool-like column.

Reply:
The only non-test place where get_bool_data is called from is in `DataFrame._reduce` which I know you've been working on recently.  The topic of how to handle object-dtype blocks has come up there, too.  If we end up handling object-dtype blocks column-wise there, that would render this distinction irrelevant.

Reply:
> IMO we should deprecate this first 

what exactly are you suggesting we deprecate? do you have an example of something that breaks on this change?

Reply:
@jorisvandenbossche can you respond here re what specific behavior you'd like to deprecate?